src/internal/serviceenv/service_env.go

```diff
@@ -296,21 +304,46 @@ func (env *NonblockingServiceEnv) initDBClient() error {
 		if err != nil {
 			return err
 		}
+		env.directDBClient = db
+		if err := prometheus.Register(sqlstats.NewStatsCollector("pg_bouncer", db.DB)); err != nil {
+			// This is not a retryable error.  Rather it will always happen for the
+			// second (and subsequent) service environment because of a naming conflict.
+			// If you see this message in production, it's a bug.  In tests, it's OK.
+			log.WithError(err).Warn("problem registering database statistics collector")
+		}
+		return db.Ping()
```

Comment:
does this prevent/reduce the pachd crashes at startup while postgres comes up? Or is etcd the limiting factor

Reply:
I didn't even think about that actually.  I added it to make sure the bad connection errors weren't happening during startup.